File: scripts/fetch_teemo.js
const fetch = require('node-fetch');
const fs = require('fs');

const RIOT_API_KEY = process.env.RIOT_API_KEY;
const BASE_URL = 'https://ddragon.leagueoflegends.com/cdn/14.8.1/data/pt_BR/champion/Teemo.json';

(async () => {
  try {
    const response = await fetch(BASE_URL);
    if (!response.ok) throw new Error('Erro ao buscar dados do Teemo');

    const json = await response.json();
    const teemoData = json.data.Teemo;

    fs.writeFileSync('data/teemo.json', JSON.stringify(teemoData, null, 2));
    console.log('Dados do Teemo salvos com sucesso.');
  } catch (err) {
    console.error('Erro ao obter dados do Teemo:', err);
    process.exit(1);
  }
})();

File: scripts/fetch_teemo.py
import requests
import json
import os

# URL da Riot para dados do Teemo (não exige API Key)
url = "https://ddragon.leagueoflegends.com/cdn/14.7.1/data/pt_BR/champion/Teemo.json"

response = requests.get(url)
if response.status_code == 200:
    teemo_data = response.json()

    os.makedirs("data", exist_ok=True)
    with open("data/teemo.json", "w", encoding="utf-8") as f:
        json.dump(teemo_data, f, ensure_ascii=False, indent=2)

    print("✅ Dados do Teemo atualizados!")
else:
    print(f"❌ Erro ao buscar dados: {response.status_code}")
